lti/views/credentials.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+
+import json
+import re
+
+from pyramid.view import view_config
+from pyramid.response import Response
+
+from lti import util
+from lti.models import OAuth2UnvalidatedCredentials
+
+
+@view_config(route_name='lti_credentials',
+             renderer='templates/lti_credentials_form.html.jinja2')
+def lti_credentials(request):
+    """Receive credentials for path A (key/secret/host)."""
+    if request.method == 'OPTIONS':
+        return cors_response(request)
+
+    credentials = util.requests.get_query_param(request, 'credentials')
+    if credentials is None:
+        return {
+            'credentials': request.registry.settings
+        }
+
+    credentials = json.loads(credentials)
+
+    request.db.add(OAuth2UnvalidatedCredentials(
+        client_id=credentials.get("path_a_key"),
+        client_secret=credentials.get("path_a_secret"),
+        authorization_server=credentials.get("path_a_host"),
+        email_address=credentials.get("path_a_email"),
+    ))
+
+    return {
+        'path_a_key': credentials.get("path_a_key"),
+        'path_a_secret': credentials.get("path_a_secret"),
+        'path_a_host': credentials.get("path_a_host"),
+        'path_a_email': credentials.get("path_a_email"),
+    }
+
+
+def cors_response(request):
```

Comment:
It might be good to have a docstring explaining what the heck this does and why. To be honest I'm kind of baffled by it. `cors_response()` is _only_ used by `lti_credentials()` when someone sends an `OPTIONS` request to the `/lti_credentials` form endpoint? In practice, does this function even get called? (If you visit the `/lti_credentials` form in a browser, and submit the form, does it any point call this function?)